hbase-common/src/main/java/org/apache/hadoop/hbase/io/ByteBuffAllocator.java

```diff
@@ -70,6 +71,8 @@
 
   public static final String MIN_ALLOCATE_SIZE_KEY = "hbase.server.allocator.minimal.allocate.size";
 
+  public static final String BYTEBUFF_ALLOCATOR_CLASS = "hbase.bytebuff.allocator.class";
```

Comment:
One suggestion is to add a class comment that it is possible to put in place an alternate bytebuffallocator by setting this param.

Second, I'd suggest adding the zeroing-out bytebuffer into this package so that it is available for test.

For example, I'd like to enable the zeroing-out bytebuffer while running a large cluster test to see if there are other places where we are prematurely releasing backing bytebuffers.

Reply:
Thanks, I'll change it.